js/llm.ts

```diff
@@ -175,27 +167,19 @@ function parseResponse(
   choiceScores: Record<string, number>
 ): Omit<Score, "name"> {
   let score = 0;
-  let error = undefined;
   const metadata: Record<string, unknown> = {};
-  try {
-    const args = JSON.parse(resp.function_call!.arguments!);
-    metadata["rationale"] = args["reasons"]?.join("\n");
-    const choice = args["choice"].trim();
-    metadata["choice"] = choice;
-    if (choiceScores[choice] !== undefined) {
-      score = choiceScores[choice];
-    } else {
-      throw new Error(`Unknown score choice ${choice}`);
-    }
-  } catch (e: unknown) {
-    score = 0;
-    error = `${e}`;
+  const args = JSON.parse(resp.function_call!.arguments!);
+  metadata["rationale"] = args["reasons"]?.join("\n");
+  const choice = args["choice"].trim();
+  metadata["choice"] = choice;
+  if (choiceScores[choice] !== undefined) {
+    score = choiceScores[choice];
+  } else {
+    throw new Error(`Unknown score choice ${choice}`);
```

Comment:
so the original motivation for swallowing errors is that, back in the day, we used to have a lot of parsing errors. Now it's quite rare. But it's technically _not_ user error when that happens.

What do we think the right thing to do is in this case? It's more like a "model failure" than a "code failure". I guess in the new code, it'll just fail this scorer and not the whole task?

Reply:
Interesting. The behavior now will be that we fail this individual scorer, and effectively log nothing or "null" for that score on this row. We'll also surface the error in the toplevel metadata. All of that seems fine. Maybe the only annoying thing is that we'll raise an error at the end of the eval saying that some scorers failed.

I'm wondering if we should just add more detail to this error message explaining that it's a model error and not a user error, or if we want to surface this kind of issue differently to the user (we could change the type of error in this case too). But I think the effect on the eval (just not logging a score and surfacing the error elsewhere) seems fine to me.

Reply:
Ok cool, null works in this case. Let's go wi th it.